### PR TITLE
[Issue #353] Publish button

### PR DESCRIPTION
**Pull request by glogiotatidis**
Convert `published` from a check-box to a button with confirmation pop-up before publishing. 

image

Will extend with permission checking

### CODE REVIEW

File: snippets/base/static/js/admin-publish.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+django.jQuery(document).ready(function() {
+    django.jQuery('[data-tool-name="publish_object"]').on("click", function(event) {
+        function confirmation() {
```
Comment (pmac) [inline snippets/base/static/js/admin-publish.js]: For my own curiosity: why did you need to put this in a nested function and then immediately call it?

Comment (glogiotatidis) [reply snippets/base/static/js/admin-publish.js]: because i suck at javascript? :D

File: snippets/base/tests/test_admin.py [hunk 2]
```diff
@@ -3,47 +3,35 @@
...
+        User.objects.create_superuser(username='admin',
+                                      email='foo@example.com',
+                                      password='admin')
+        self.client.login(username='admin', password='admin')
+
+    def test_action_publish_snippet(self):
+        snippet = SnippetFactory(published=False)
+        self.assertFalse(snippet.published)
+        snippet_action_url = '/admin/base/snippet/{}/actions/publish_object/'.format(snippet.pk,)
+        self.client.get(snippet_action_url)
```
Comment (pmac) [inline snippets/base/tests/test_admin.py]: Hmm.. didn't notice this before, but it's odd that the action can be taken using a GET request. Should really be a POST to modify an object.

Comment (glogiotatidis) [reply snippets/base/tests/test_admin.py]: That's a very good point and I'm immediately returning this to WIP. It seems that this is how django-object-actions implements things. I'll re-think the approach

**Comment by glogiotatidis** [comment]
I'm going to close this one as I don't like the approach and while related it's not a blocker to tackle #353 

Thanks for the insightful info @pmac

**Comment by pmac** [comment]
That's a bummer. I was thinking how useful a project object-actions was. I don't think it would take a huge change to fix this GET/POST issue, but I could easily be wrong.

**Comment by glogiotatidis** [comment]
> That's a bummer. I was thinking how useful a project object-actions was. I don't think it would take a huge change to fix this GET/POST issue, but I could easily be wrong.

They have a couple related issues open, maybe I'll take a look later on.
